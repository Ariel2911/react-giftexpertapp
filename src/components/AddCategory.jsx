import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
    // console.log('handleInputValue llamado');
  };

  const handleSubmit = (e) => {
    console.log('handleSubmit llamado');
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={(e) => handleInputValue(e)}
      />
      <button>Add</button>
    </form>
  );
};

AddCategory.protoType = {
  setCategories: PropTypes.func.isRequired,
};
