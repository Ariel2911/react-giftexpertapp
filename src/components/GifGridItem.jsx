export const GifGridItem = ({ id, title, url }) => {
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        border: '1px solid black',
        borderRadius: '8px',
        margin: '4px',
        width: '150px',
        height: '160px',
      }}
    >
      <img src={url} alt={title} style={{ width: '100%', height: '75%' }} />
      <p
        style={{
          textAlign: 'center',
          height: '25%',
          margin: '0',
          overflow: 'hidden',
        }}
      >
        {title}
      </p>
    </div>
  );
};
