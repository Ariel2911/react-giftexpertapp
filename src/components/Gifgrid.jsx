import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <li style={{ display: 'flex', flexWrap: 'wrap' }}>
        {loading ? (
          <p>Cargando</p>
        ) : (
          images.map((image) => {
            return <GifGridItem key={image.id} {...image} />;
          })
        )}
      </li>
    </>
  );
};
