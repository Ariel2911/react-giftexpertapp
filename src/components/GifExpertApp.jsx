import { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './Gifgrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState( defaultCategories );

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};
