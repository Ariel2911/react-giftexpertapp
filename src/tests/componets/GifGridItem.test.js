import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Prueba del componete GifGridItem', () => {
    const title = 'Un Titulo';
    const url = 'http://localhost/algo.jpg';
    let wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe contener', () => {
        const textValue = wrapper.find('p').text().trim();
        
        expect(textValue).toBe(title);
    });

    test('Pruebas de los atributos de img', () => {

        const imgSrc = wrapper.find('img').prop('src');
        console.log(imgSrc) 
        const imgAlt = wrapper.find('img').prop('alt'); 

        expect(imgSrc).toBe(url);
        expect(imgAlt).toBe(title);
    });

    test('Pruebas de los estylos del div', () => {
        const width = '150px';
        const styledDiv = wrapper.find('div').prop('style').width;
        
        expect(styledDiv).toBe(width);
    })

});