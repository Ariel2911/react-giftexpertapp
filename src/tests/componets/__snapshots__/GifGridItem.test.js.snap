// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componete GifGridItem Debe de mostrar el componente correctamente 1`] = `
<div
  style={
    Object {
      "border": "1px solid black",
      "borderRadius": "8px",
      "display": "flex",
      "flexDirection": "column",
      "height": "160px",
      "margin": "4px",
      "width": "150px",
    }
  }
>
  <img
    alt="Un Titulo"
    src="http://localhost/algo.jpg"
    style={
      Object {
        "height": "75%",
        "width": "100%",
      }
    }
  />
  <p
    style={
      Object {
        "height": "25%",
        "margin": "0",
        "overflow": "hidden",
        "textAlign": "center",
      }
    }
  >
    Un Titulo
  </p>
</div>
`;
